benchmark_test.go

```diff
@@ -0,0 +1,124 @@
+package jasper
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/send"
+	"github.com/stretchr/testify/require"
+)
+
+type procMaker func(context.Context, *CreateOptions) (Process, error)
+type logMaker func() Logger
+type procMap map[string]procMaker
+type timeMap map[string]time.Duration
+
+func yesCreateOpts(timeout time.Duration) CreateOptions {
+	return CreateOptions{Args: []string{"yes"}, Timeout: timeout}
+}
+
+func testDurations() timeMap {
+	return timeMap{
+		"Send1Second":  1 * time.Second,
+		"Send2Seconds": 2 * time.Second,
+		"Send4Seconds": 4 * time.Second,
+		"Send8Seconds": 8 * time.Second,
+	}
+}
+
+func getProcMap() procMap {
+	return procMap{
+		"Blocking": newBlockingProcess,
+		"Basic":    newBasicProcess,
+	}
+}
+
+func runBenchmark(ctx context.Context, b *testing.B, makeProc procMaker, optsList []*CreateOptions) {
+	for n := 0; n < b.N; n++ {
+		proc, err := makeProc(ctx, optsList[n])
+		require.NoError(b, err)
+		_ = proc.Wait(ctx)
+	}
+}
+
+func makeAllCreateOpts(iters int, timeout time.Duration, logger Logger) []*CreateOptions {
+	optsList := make([]*CreateOptions, iters)
+	for n := 0; n < iters; n++ {
+		opts := yesCreateOpts(timeout)
+		optsList[n] = &opts
+		optsList[n].Output.Loggers = []Logger{logger}
+	}
+	return optsList
+}
+
+func BenchmarkFileLogger(b *testing.B) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for procName, makeProc := range getProcMap() {
+		b.Run(procName, func(b *testing.B) {
+			for sendTime, timeout := range testDurations() {
+				b.Run(sendTime, func(b *testing.B) {
+					file, err := ioutil.TempFile("build", "bench_out.txt")
+					require.NoError(b, err)
+					defer os.Remove(file.Name())
```

Comment:
I think this means that we're timing the amount of time required to delete the file. 

would it make more sense to create a temporary directory, and then use remove all for the entire directory outside of the scope of the benchmark

Reply:
Nothing is really being timed in the benchmark. The amount of time the process is given depends only on the timeout value, so the testing.B timer isn't used really.